src/Bundle/Tests/DependencyInjection/Compiler/RegisterFQCNControllersPassTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ResourceBundle\Tests\DependencyInjection\Compiler;
+
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractCompilerPassTestCase;
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Compiler\RegisterFQCNControllersPass;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+final class RegisterFQCNControllersPassTest extends AbstractCompilerPassTestCase
+{
+    /**
+     * @test
+     * @doesNotPerformAssertions
+     */
+    public function it_throws_exception_if_resource_does_not_exist(): void
+    {
+        $this->compile();
+
+        $this->throwException(new InvalidArgumentException());
+    }
```

Comment:
Are you sure about this test?

Reply:
I hope, that it's testing this part of code:

> try {
            $resources = $container->getParameter('sylius.resources');
        } catch (InvalidArgumentException $exception) {
            return;
        }

Reply:
What about the check for the second exception that is thrown in the `validateSyliusResource` method?

Reply:
Maybe we should also check if there is any service registered in the container? I don't know how, to be honest, as we do not set any `sylius.resources` parameter (to check if the exception is thrown) 🤷‍♂️ xD

Reply:
I didn't dig into it too deeply, but according to https://github.com/sebastianbergmann/phpunit-documentation/issues/171 this is a proper way to do it. Just `throwException` method can be removed.

Reply:
I think it doesn't work, since we're extending `AbstractCompilerPassTestCase`, we have this test automatically:

```php
/**
 * This test will run the compile method.
 *
 * @test
 * @coversNothing
 */
final public function compilation_should_not_fail_with_empty_container(): void
{
    try {
        $this->compile();

        // Ensure assertion for strict mode
        $this->assertFalse($this->container->has('no_service'));
    } catch (\Exception $e) {
        $this->fail('The compiler pass should not fail with an empty container.');
    }
}
```

Compiler pass shouldn't throw an exception with an empty container.

Reply:
Anyway, what you wanted to do is `$this->expectException()`.